tests/test_home/TestControllers.py

```diff
@@ -0,0 +1,31 @@
+import unittest
+from database import create_session
+from mock import mock
+
+import mod_home.controllers as home
+from mod_home.models import GeneralData, CCExtractorVersion
+
+
+def mock_database(db_string, drop_tables=False):
+    db = create_session(db_string, drop_tables)
+    mock_db = mock.MagicMock(spec=db)
+    return mock_db
+
+
+class TestControllers(unittest.TestCase):
+
+    def setUp(self):
+        with mock.patch('database.create_session') as create_session:
```

Comment:
Instead of mocking the entire database layer, I'd probably initially add an in-memory database and depending on the test, inject some interesting data in it to test the actual rendering of templates.

You can take a look at https://github.com/canihavesomecoffee/ci_workshop for this :+1:

Reply:
ok